collector/monitors/delegations_distribution_monitor.go

```diff
@@ -0,0 +1,133 @@
+	"sync"
+
+	"github.com/lidofinance/terra-monitors/internal/math"
+
+	"github.com/lidofinance/terra-monitors/collector/config"
+	"github.com/lidofinance/terra-monitors/collector/types"
+	"github.com/lidofinance/terra-monitors/internal/client"
+	terraClient "github.com/lidofinance/terra-monitors/openapi/client"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	DelegationsDistributionImbalance MetricName = "delegations_distribution_imbalance"
+)
+
+type DelegationsDistributionMonitor struct {
+	metrics              map[MetricName]MetricValue
+	metricVectors        map[MetricName]*MetricVector
+	apiClient            *terraClient.TerraLiteForTerra
+	validatorsRepository ValidatorsRepository
+	nMads                int64
+	logger               *logrus.Logger
+	lock                 sync.RWMutex
+}
+
+func NewDelegationsDistributionMonitor(
+	cfg config.CollectorConfig,
+	logger *logrus.Logger,
+	repository ValidatorsRepository,
+) *DelegationsDistributionMonitor {
+	m := &DelegationsDistributionMonitor{
+		metrics:              make(map[MetricName]MetricValue),
+		metricVectors:        make(map[MetricName]*MetricVector),
+		apiClient:            client.New(cfg.LCD, logger),
+		validatorsRepository: repository,
+		nMads:                cfg.DelegationsDistributionConfig.NumMedianAbsoluteDeviations,
+		logger:               logger,
+		lock:                 sync.RWMutex{},
+	}
+
+	m.InitMetrics()
+
+	return m
+}
+
+func (m *DelegationsDistributionMonitor) Name() string {
+	return "DelegationsDistribution"
+}
+
+func (m *DelegationsDistributionMonitor) providedMetrics() []MetricName {
+	return []MetricName{}
+}
+
+func (m *DelegationsDistributionMonitor) providedMetricVectors() []MetricName {
+	return []MetricName{
+		DelegationsDistributionImbalance,
+	}
+}
+
+func (m *DelegationsDistributionMonitor) InitMetrics() {
+	initMetrics(m.providedMetrics(), m.providedMetricVectors(), m.metrics, m.metricVectors)
+}
+
+func (m *DelegationsDistributionMonitor) Handler(ctx context.Context) error {
+	// tmp* for 2stage nonblocking update data
+	tmpMetrics := make(map[MetricName]MetricValue)
+	tmpMetricVectors := make(map[MetricName]*MetricVector)
+	initMetrics(m.providedMetrics(), m.providedMetricVectors(), tmpMetrics, tmpMetricVectors)
+
+	validatorsInfo, err := m.getValidatorsInfo(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to getValidatorsInfo: %w", err)
+	}
+
+	var delegations []*big.Int
+	for _, validatorInfo := range validatorsInfo {
+		delegations = append(delegations, validatorInfo.Tokens)
+	}
+
+	var outliers = math.GetMeanAbsoluteDeviationOutliers(delegations, m.nMads)
+	for _, outlierIdx := range outliers {
+		//tmpMetricVectors[DelegationsDistributionImbalance].Set(validatorsInfo[outlierIdx].Address, 1)
+		tmpMetricVectors[DelegationsDistributionImbalance].Set(
+			fmt.Sprintf("%s (%s)", validatorsInfo[outlierIdx].Address, validatorsInfo[outlierIdx].Moniker), 1)
+	}
+
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	copyMetrics(tmpMetrics, m.metrics)
```

Comment:
Unused variable? tmpMetrics is empty and not used anywhere except this copy function

Reply:
fixed